### PR TITLE
Issue #110: Added documentation to SFCGAL::algorithm::extrude

### PR DESCRIPTION
**Pull request by danielcu888**
* Extended extrude function descriptions to indicate how
  Geometries that contain measures that are input into
  extrude will generate results with thise measures stripped.

### CODE REVIEW

File: src/algorithm/extrude.h [hunk 1]
```diff
@@ -13,56 +13,132 @@
...
+// Class forward declarations.
 struct NoValidityCheck;
 
 /**
- * extrude a Geometry with a direction
- * @pre g is a valid geometry
+ * @brief Returns a Geometry equal to the specified Geometry,
+ *   extruded by the specified displacement.
+ * @param g The specified Geometry.
+ * @param dx The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:41]:
```suggestion
 * @param dx The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 2]
```diff
@@ -13,56 +13,132 @@
...
+ *   the x-direction.
+ * @param dy The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:43]:
```suggestion
 * @param dy The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 3]
```diff
@@ -13,56 +13,132 @@
...
+ *   the y-direction.
+ * @param dz The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:45]:
```suggestion
 * @param dz The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 4]
```diff
@@ -13,56 +13,132 @@
...
+                                              ) ;
 
 /**
- * extrude a Geometry with a direction
- * @pre g is a valid geometry
- * @pre dx, dy and dz are finite values
+ * @brief Returns a Geometry equal to the specified Geometry,
+ *   extruded by the specified displacement.
+ * @param g The specified Geometry.
+ * @param dx The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:96]:
```suggestion
 * @param dx The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 5]
```diff
@@ -13,56 +13,132 @@
...
+ *   the x-direction.
+ * @param dy The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:98]:
```suggestion
 * @param dy The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 6]
```diff
@@ -13,56 +13,132 @@
...
+ *   the y-direction.
+ * @param dz The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:100]:
```suggestion
 * @param dz The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 7]
```diff
@@ -13,56 +13,132 @@
...
+                                              );
 
 /**
- * @brief extrude a Geometry by a given vector
- * @todo Improve 3D surface extrude (Extrude only faces whose dot( v, normal ) > 0 and use Polyhedron union to
- *         get output geometries with a clean topology)
+ * @brief Returns a Geometry equal to the specified Geometry,
+ *   extruded by the specified displacement vector.
+ * @param g The specified Geometry.
+ * @param v The specified dispalcement vector.
```
Comment (sloriot) [inline src/algorithm/extrude.h:123]:
```suggestion
 * @param v The specified displacement vector.
```

File: src/algorithm/extrude.h [hunk 8]
```diff
@@ -13,56 +13,132 @@
...
+                                              , Kernel::FT dz
+                                              ) ;
 
 /**
- * extrude a Geometry with a direction
- * @pre g is a valid geometry
+ * @brief Returns a Geometry equal to the specified Geometry,
+ *   extruded by the specified displacement.
+ * @param g The specified Geometry.
+ * @param dx The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:67]:
```suggestion
 * @param dx The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 9]
```diff
@@ -13,56 +13,132 @@
...
+ *   the x-direction.
+ * @param dy The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:69]:
```suggestion
 * @param dy The component of the specified displacement in
```

File: src/algorithm/extrude.h [hunk 10]
```diff
@@ -13,56 +13,132 @@
...
+ *   the y-direction.
+ * @param dz The component of the specified dispalcement in
```
Comment (sloriot) [inline src/algorithm/extrude.h:71]:
```suggestion
 * @param dz The component of the specified displacement in
```

File: src/algorithm/extrude.h (hunk 1 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 3 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 2 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 4 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 5 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 6 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 8 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 10 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 9 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

File: src/algorithm/extrude.h (hunk 7 above)
Comment (danielcu888) [reply src/algorithm/extrude.h]: Fixed

**Comment by danielcu888** [comment]
@lbartoletti I have conducted the remedial.

**Comment by danielcu888** [comment]
> Note the since CGAL 4.13 there is an [`extrude_mesh()`](https://doc.cgal.org/4.13.2/Polygon_mesh_processing/group__PMP__meshing__grp.html#ga102e1ba5d709c5e53e013160792e4d08) function in CGAL.

@sloriot, regarding extrude_mesh() perhaps create an improvement ticket to re-implement the existing functionality.

**Comment by lbartoletti** [comment]
Thank you daniel.

**Comment by danielcu888** [comment]
You're welcome @lbartoletti

**Pull request merged**